src/utils/web3.ts

```diff
@@ -0,0 +1,21 @@
+import { _TypedDataEncoder } from 'ethers/lib/utils'
+import type { EIP712TypedData } from '@gnosis.pm/safe-react-gateway-sdk'
+import type { TypedDataDomain } from 'ethers'
+
+import { getWeb3 } from '@/hooks/wallets/web3'
+
+export const signTypedData = async ({ domain, types, message }: EIP712TypedData): Promise<string> => {
```

Comment:
maybe passing the signer object from outside is better, then the function would be easier to test and you don't need to handle the case when someone invokes it while some global var is unavailable